sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6500,15 +6504,6 @@ pi_result piEnqueueEventsWaitWithBarrier(pi_queue Queue,
     for (auto It = Queue->CommandListMap.begin();
          It != Queue->CommandListMap.end(); ++It)
       CmdLists.push_back(It);
-  } else if (Queue->ComputeQueueGroup.ZeQueues.empty() &&
-             Queue->CopyQueueGroup.ZeQueues.empty()) {
-    // If there are no queues, we get any available command list.
-    pi_command_list_ptr_t CmdList;
-    if (auto Res = Queue->Context->getAvailableCommandList(
-            Queue, CmdList,
-            /*UseCopyEngine=*/false, OkToBatch))
-      return Res;
-    CmdLists.push_back(CmdList);
```

Comment:
I think we should run this code is the loop in https://github.com/intel/llvm/pull/7516/files#diff-15dd1eb076d2164bd9e87d9057f05f652a716498e8cdf5975e564c65309a0985R6520-R6526 comes empty. So still create a command-list and insert the barrier there (and the *Event will always be returned to the user)

Reply:
If there are no ZeQueues then that means there have been no submissions on that queue. So why add a barrier when we know there is nothing to wait for?

Reply:
Because users wanted us to insert a barrier and get them back an event. Alternatively, we could create an empty event, signal it, and return.

Reply:
Ok, now I understand.